test/access-control.integration.js

```diff
@@ -324,4 +332,23 @@ describe('access control - integration', function() {
       return '/api/accounts-updating/' + this.accountWithReplaceOnPUTfalse.id + '/replace';
     }
   });
+
+  describe('/accounts with upsertWithWhere', function() {
+    var count = 0;
+    before(function() {
+      var roleModel = loopback.getModelByType(loopback.Role);
+      roleModel.registerResolver('$dummy', function(role, context, callback) {
+        process.nextTick(function() {
+          if (context.remotingContext) {
+            count++;
+          }
+          callback && callback(null, false); // Always true
+        });
+      });
+    });
+    lt.beforeEach.givenModel('accountWithReplaceOnPUTtrue');
```

Comment:
`accountWithReplaceOnPUTtrue` is added specifically for testing `replace` methods when `replaceOnPUT` flag is set to true; I don't think you should test with that model; I believe you should test it with `account` model instead to make the purpose of test clear.